### PR TITLE
Add aten::replication_pad1d/2d/3d and backward

### PR DESCRIPTION
**Pull request by majing921201**

### CODE REVIEW

File: src/ATen/native/xpu/ReplicationPadding.cpp [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+}
+
+Tensor& XPUNativeFunctions::replication_pad2d_backward_out(
+    const Tensor& grad_output,
+    const Tensor& input,
+    IntArrayRef padding,
+    Tensor& grad_input) {
+  // See Note [Writing Nondeterministic Operations]
+  // Nondeterministic because of atomicAdd usage
+  globalContext().alertNotDeterministic("replication_pad2d_backward_out_xpu");
```
Comment (fengyuan14) [inline src/ATen/native/xpu/ReplicationPadding.cpp:265]: Move to kernel level.

File: src/ATen/native/xpu/ReplicationPadding.cpp [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+  return grad_input;
+}
+
+Tensor XPUNativeFunctions::replication_pad2d_backward(
+    const Tensor& grad_output,
+    const Tensor& input,
+    IntArrayRef padding) {
+  // See Note [Writing Nondeterministic Operations]
+  // Nondeterministic because of atomicAdd usage
+  globalContext().alertNotDeterministic("replication_pad2d_backward_xpu");
```
Comment (fengyuan14) [inline src/ATen/native/xpu/ReplicationPadding.cpp:277]: ditto

File: src/ATen/native/xpu/ReplicationPadding.cpp [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+}
+
+Tensor& XPUNativeFunctions::replication_pad3d_backward_out(
+    const Tensor& grad_output,
+    const Tensor& input,
+    IntArrayRef padding,
+    Tensor& grad_input) {
+  // See Note [Writing Nondeterministic Operations]
+  // Nondeterministic because of atomicAdd usage
+  globalContext().alertNotDeterministic("replication_pad3d_backward_out_xpu");
```
Comment (fengyuan14) [inline src/ATen/native/xpu/ReplicationPadding.cpp:322]: ditto

File: src/ATen/native/xpu/sycl/ReplicationPaddingKernels.cpp [hunk 4]
```diff
@@ -0,0 +1,737 @@
...
+
+  sycl_kernel_submit(
+      sycl::range<3>(nbatch, nplane, work_group_size * work_group_num),
+      sycl::range<3>(1, 1, work_group_size),
+      queue,
+      kfn);
+}
+
+template <typename scalar_t>
+struct replication_pad1d_forward_kernel_functor {
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/ReplicationPaddingKernels.cpp:89]: Please rename all kernel related structures,
1. `KernelFunctor` is used to name kernel frame.
2. The scalar functor is named by like `ReplicationPad1dForwardFunctor`. W/o kernel in the name.

File: test/xpu/extended/run_test_with_skip.py [hunk 5]
```diff
@@ -163,6 +163,12 @@
     "test_compare_cpu_std_mean_xpu_bfloat16",
     "test_compare_cpu_sub_xpu_float16",
     "test_compare_cpu_var_mean_xpu_bfloat16",
+
+    # Precision error.
+    # Mismatched elements: 2 / 125 (1.6%)
+    # Greatest absolute difference: 0.001953125 at index (2, 0, 0) (up to 0.001 allowed)
+    # Greatest relative difference: 0.007568359375 at index (2, 0, 0) (up to 0.001 allowed)
+    "test_compare_cpu_cumprod_xpu_bfloat16",
```
Comment (fengyuan14) [inline test/xpu/extended/run_test_with_skip.py:171]: Do we still need skip it? We can pass in recent pre-ci?

Comment (majing921201) [reply test/xpu/extended/run_test_with_skip.py:171]: > Do we still need skip it? We can pass in recent pre-ci?

I enabled this test case after adding missing "," in the pr, now the failure is stable occurrence

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**